src/app/shared/services/backend/link-preview.service.ts

```diff
@@ -0,0 +1,53 @@
+import {Injectable} from '@angular/core'
+import {HttpClient, HttpParams} from '@angular/common/http'
+import {Observable} from 'rxjs'
+import {environment} from '../../../../environments/environment'
+import {map} from "rxjs/operators"
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkPreviewService {
+  path = `${environment.backendURL}/api/link`
+
+  constructor(private http: HttpClient) {
+  }
+
+  previewLink(url: string): Observable<LinkPreviewResponse> {
+    const params = new HttpParams().set('url', url)
+    return this.http.get<LinkPreviewResponse>(`${this.path}/preview`, {params: params}).pipe(
+      map((response) => {
+        if (response.open_graph?.image?.url) {
+          return {
+            url: response.url,
+            open_graph: {
+              ...response.open_graph,
+              image: {
+                ...response.open_graph.image,
+                url: response.open_graph.image.url.replace(/&amp;/, '&')
+              }
+            }
+          }
+        }
+        return response
```

Comment:
Ok, this logic could me more complicated because you can have various combinations on your way; with or without image, with or without title, with or without description.
If we have the picture, we should show the picture,
If we have the title, we should show the title,
If we have the description, we should show description,
If we don't have anything from the opengraph response, or our service returned an error, we should just show the url that we have.

I recommend you to find broken links without opengraph metatags so you can test these cases.

Reply:
I already do this conditional display here:
(<info-icon><title>) or (<info-icon><url>) if title is not available)
<image> if available
<description> if available
this is done in the offers HTML file:
```html
<a href="{{ link.url }}" class="ml-3 text-sm" target="_blank" rel="noopener noreferrer">
    {{ link.open_graph?.title ?? link.url }}
</a>
<div *ngIf="link.open_graph?.image?.url" class="flex justify-center p-2">
<img class="max-h-60"
     src="{{ link.open_graph!.image!.url }}"
     alt="News item photo" srcset="">
</div>
<div *ngIf="link.open_graph?.description" class="flex p-2 text-sm">
    {{ link.open_graph!.description }}
</div>
```

Reply:
I needed to add `.replace(/&amp;/, '&')` for image URL because our link service returns it with ampersand encoded as `&amp;` and that caused image not to be fetchable.

Reply:
I agree, but `&` is not the only case where it would break. Consider using [decodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) and [decodeURI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURI) to properly handle those links. You can create these methods inside a service, no need to extract it elsewhere as we'll do it if we need.

Reply:
Hmm the problem here is that `&amp;` is NOT URI encoded, it's HTML encoded... I will talk to Eugen and see why does the backend send it like that.

Reply:
Checked with Eugen, it looks like the link preview service is working as expected. For some reason, Reddit's meta tag has a link with `&amp;` in it. What do you think would be the best solution here, use some sort of HTML decoding for image URL?

Reply:
https://stackoverflow.com/questions/3311705/should-meta-tags-be-html-encoded

It looks like content inside metatag should always be HTML encoded. Therefore, somebody needs to decode it, either backend or frontend (every single field). 
My proposal is, if we already have a backend service to do that, it would be great that backend does this job because, once that we have a JSON response, the stuff inside JSON should not be HTML encoded.

For now, I'll take care of some cases, but it affects all fields in opengraph object of the backend response.

@githinho

Reply:
Fixed in the version of `link-preview-service`